contracts/ExitHandler.sol

```diff
@@ -0,0 +1,202 @@
+
+/**
+ * Copyright (c) 2018-present, Leap DAO (leapdao.org)
+ *
+ * This source code is licensed under the Mozilla Public License, version 2,
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+pragma solidity 0.4.24;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+
+import "./DepositHandler.sol";
+import "./Bridge.sol";
+import "./TxLib.sol";
+import "./PriorityQueue.sol";
+
+contract ExitHandler is DepositHandler {
```

Comment:
This looks strange from OOP point of view

Reply:
How so?

Reply:
OOP would be something like:
`Vault <- DepositableVault <- ExitableVault`
?

Reply:
Since we have multiple inheritance it can be `Vault ← DepositableVault & ExitableVault`. But it's really a nitpicking :-) I know too few about solidity to have a voice here

Reply:
Makes sense. The most reasonable relationship would be to have a composition I guess, that is a Vault has a DepositHandler and it has an ExitHandler.

Reply:
But we will need a separate exit procedure for unincluded deposits so we will need a version of ExitHandler that is aware of the deposits.

Reply:
As i said, you can ignore me here :-)